engine/src/main/java/org/hibernate/search/engine/search/highlighter/dsl/SearchHighlighterFastVectorHighlighterOptionsStep.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.engine.search.highlighter.dsl;
+
+import java.util.Collection;
+
+/**
+ * The step in a fast vector highlighter definition where options can be set. Refer to your particular backend documentation
+ * for a more detailed information on the exposed settings.
+ */
+public interface SearchHighlighterFastVectorHighlighterOptionsStep
+		extends SearchHighlighterOptionsStep<SearchHighlighterFastVectorHighlighterOptionsStep> {
+
+	/**
+	 * Specify how far to scan for {@link #boundaryChars(String) boundary characters} when
+	 * a {@link SearchHighlighterBoundaryScannerTypeFastVectorHighlighterOptionsStep#chars() characters oundary scanner}  is used.
+	 *
+	 * @param max The number of characters.
+	 * @return The next step in a highlighter definition.
+	 */
+	SearchHighlighterFastVectorHighlighterOptionsStep boundaryMaxScan(int max);
+
+	/**
+	 * Specify a string of characters to look for when scanning for boundaries when
+	 * a {@link SearchHighlighterBoundaryScannerTypeFastVectorHighlighterOptionsStep#chars() characters oundary scanner}  is used.
+	 *
+	 * @param boundaryChars The string of boundary characters.
+	 * @return The next step in a highlighter definition.
+	 */
+	SearchHighlighterFastVectorHighlighterOptionsStep boundaryChars(String boundaryChars);
+
+	/**
+	 * Specify the margin from which to start highlighting.
+	 *
+	 * @param offset The margin to be applied to a fragment list builder of a fast vector highlighter.
```

Comment:
Or... or... we add `border` to the mix and make it as scary as the CSS and nobody would want to use this setting 😄🙈.

Jokes aside 😄... this confusing mix comes from the ES... they've named the property as an `offset` but explained it as a margin .... then what it really does is ... :
```java
int startOffset = 0;
...
while ( has fragments )
...
int currentPhraseStartOffset = phraseInfo.getStartOffset();
int spanStart = Math.max(currentPhraseStartOffset - margin, startOffset);
int spanEnd = Math.max(currentPhraseEndOffset, spanStart + fragCharSize);
// ..... some code not using spanStart
// then the start is overridden when a span is added to the frament list.
spanStart = currentPhraseStartOffset - newMargin;
if (spanStart < startOffset) {
    spanStart = startOffset;
}
...
```
https://github.com/apache/lucene/blob/main/lucene/highlighter/src/java/org/apache/lucene/search/vectorhighlight/BaseFragListBuilder.java#L71-L103
Sadly, there seem to be not much docs on the Lucene side... and from the code, it looks like it really only may influence a `spanEnd` which would define how many phrases will be picked per span

Reply:
Ok; if you manage to make sense of that stuff (and to make it work consistently) we can keep it, but otherwise I'd say let's just not expose the feature and wait for someone to request it, with an explanation of their use case and of how they think that feature is supposed to work?

Reply:
(I'm suggesting that because I think there are various weird results related to this feature in tests, too)

Reply:
I've ended up removing this property... I couldn't get consistent results out of it. After experimenting and debugging the `BaseFragListBuilder`. Since that maring parameter has a constraint `minFragCharSize = Math.max(1, margin * MIN_FRAG_CHAR_SIZE_FACTOR);` changing the margin to be bigger requires increasing the fragment size. And if one would only increase the fragment size, that would already lead to more text being included in the resulting fragment. Then if you highlight something at the very beginning of the sentence (assuming that in such case, you'd want to have a margin catch a previous sentence as well), it still won't need the margin as there is a "recentering the results" that would recompute start/end of the span and do that anyway... 
I've sent a question to lucne mailing list to see if anyone could provide some insights into what that margin was supposed to actually do 😃